docs/source/mods/line-optimization.rst

```diff
@@ -0,0 +1,243 @@
+          >>> linepath_data.head(4)
+            linename	edgeSource	edgeTarget
+            0	new7_B	1	2
+            1	new7_B	2	6
+            2	new7_B	6	8
+            3	new7_B	8	6
+          >>> demand_data.head(4)
+            source	target	demand
+            0	1	2	5
+            1	1	3	5
+            2	1	4	25
+            3	1	5	10
+          >>> frequencies = [1,3]
+
+      For the example we used data of the Sioux-Falls network. It is not considered as a realistic one. However,
+      this network can be found on different websites when considering traffic problems
+      (originally by Hillel Bar-Gera http://www.bgu.ac.il/~bargera/tntp/). We added a set of line routes.
+      Note that the output shown above contains some additional information that is not required for computation, for example
+      the property length in the edge data.
+      Also, ``posx`` and ``posy`` in the ``node_data`` is not used for computation. But it can be used to visualize the
+      network as done below.
+      It is important that all data is consistant. For example, ``edgeSource``, ``edgeTarget``
+      in the ``linepath_data`` must correspond to a ``number`` in the node_data. The same holds
+      for ``source`` and ``target`` in ``edge_data`` and ``demand_data``.
+      In the code it is checked that all tables provide the relevant columns.
+      Note that the edges are assumed to be directed and both direction need to be defined if an edge
+      can be traversed in both directions. In the same way, a line is a directed path. If a line is
+      turning at the end point and goes back the same way, the nodes need to be added again in reverse order.
+
+Solution
+--------
+
+The solution consists of two information
+
+- the total cost of the optimal line plan
+- the optimal line plan as a list of linename-frequency tuples.
+
+The strategy can be defined via a Boolean parameter **shortestPaths**. This parameter has a default value (True)
+which uses approach 1, i.e., routing passengers on shortest paths only.
+Note that strategy 1 needs the python package networkx. If this is not available, the second approach is used.
+The second approach is also used if the parameter shortestPaths is set to False.
+
+.. tabs::
+
+  .. group-tab:: pandas
+
+      .. doctest:: solve
+          :options: +NORMALIZE_WHITESPACE
+
+          >>> from gurobi_optimods import datasets
+          >>> from gurobi_optimods.line_optimization import line_optimization
+          >>> node_data, edge_data, line_data, linepath_data, demand_data = datasets.load_siouxfalls_network_data()
+          >>> frequencies = [1,3]
+          >>> objCost, finalLines = line_optimization(node_data, edge_data, line_data, linepath_data, demand_data, frequencies, True, verbose=False)
+          >>> objCost
+          211.0
+          >>> finalLines
+          [('new271_B', 1), ('new31_B', 1), ('new407_B', 1), ('new415_B', 3), ('new423_B', 3), ('new535_B', 3), ('new551_B', 3), ('new71_B', 1)]
```

Comment:
Can you please change the variable names to use snake case? This is more the pythonic style, i.e. `obj_cost` and `final_lines`.